uniflow/op/model/model_server.py

```diff
@@ -248,41 +248,62 @@ class HuggingfaceModelServer(AbsModelServer):
 
     def __init__(self, model_config: Dict[str, Any]) -> None:
         # import in class level to avoid installing transformers package
-        from transformers import pipeline  # pylint: disable=import-outside-toplevel
+        super().__init__(model_config)
+        self._model_config = HuggingfaceModelConfig(**self._model_config)
+        if self._model_config.neuron is False:
+            try:
+                from transformers import (  # pylint: disable=import-outside-toplevel
+                    pipeline,
+                )
+            except ModuleNotFoundError as exc:
+                raise ModuleNotFoundError(
+                    "Please install transformers to use HuggingfaceModelServer. You can use `pip install transformers` to install it."
+                ) from exc
+            model, tokenizer = self._get_model()
+            # explicitly set batch_size for pipeline
+            # for batch inference.
+            self._pipeline = pipeline(
+                "text-generation",
+                model=model,
+                tokenizer=tokenizer,
+                device_map="auto",
+                max_new_tokens=768,
+                num_return_sequences=1,
+                repetition_penalty=1.2,
+                eos_token_id=tokenizer.eos_token_id,
+                pad_token_id=tokenizer.pad_token_id,
+                batch_size=self._model_config.batch_size,
+            )
+        else:
+            from uniflow.op.model.neuron_utils import (  # pylint: disable=import-outside-toplevel
+                Neuron,
+            )
+
+            model, tokenizer = Neuron.get_neuron_model(
+                self._model_config.model_name, self._model_config.batch_size
+            )
+            self._pipeline = partial(
+                Neuron.neuron_infer, model=model, tokenizer=tokenizer
+            )
```

Comment:
qq: is there a performance issue for not using `partial` but directly call model, tokenizer, and input value during run?

Reply:
This is to simplify the __call__ method.